integrations/cdi/jpa-cdi/src/main/java/io/helidon/integrations/cdi/jpa/PersistenceExtension.java

```diff
@@ -298,6 +299,23 @@ private <T> void makePersistencePropertyARepeatableQualifier(@Observes BeforeBea
         event.addQualifier(PersistenceProperty.class);
     }
 
+    /**
+     * {@linkplain ProcessBeanAttributes#veto() Vetoes} any bean whose bean types includes the deprecated {@link
+     * JtaDataSourceProvider} class, since it is replaced by {@link JtaAdaptingDataSourceProvider}.
+     *
+     * @param event the {@link ProcessBeanAttributes} event in question; must not be {@code null}
+     *
+     * @exception NullPointerException if {@code event} is {@code null}
+     *
+     * @see JtaAdaptingDataSourceProvider
+     */
+    private void vetoDeprecatedJtaDataSourceProvider(@Observes ProcessBeanAttributes<JtaDataSourceProvider> event) {
+        if (!this.enabled) {
+            return;
+        }
+        event.veto();
```

Comment:
If a developer includes a type with the deprecated class which we veto, should we log a warning or info message?

Reply:
Ever so slightly tricky: this will only be the case so long as the old and the new stuff coexist, and there's nothing the user can do about it. I'll split the difference and log at the `FINE` level.